wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -148,23 +152,46 @@ private fun parseFileDescriptor(fileDescriptor: DescriptorProtos.FileDescriptorP
   )
 }
 
-private fun parseMessage(path: List<Int>, helper: SourceCodeHelper, message: DescriptorProtos.DescriptorProto, descs: Plugin.DescriptorSource): MessageElement {
-  val nestedTypes = mutableListOf<TypeElement>()
+private fun parseEnum(path: List<Int>, helper: SourceCodeHelper, enum: DescriptorProtos.EnumDescriptorProto, descs: DescriptorSource): EnumElement {
+  val info = helper.getLocation(path)
+  val constants = mutableListOf<EnumConstantElement>()
+  for (enumValueDescriptorProto in enum.valueList) {
+    constants.add(EnumConstantElement(
+      location = info.loc,
```

Comment:
`info` is for the enum, not the values inside it. You need to do something like what is done for nested messages: you must append `VALUE_FIELD_NUMBER` and then the index to the path and fetch that location from the helper.